cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexCreator.java

```diff
@@ -79,7 +79,7 @@ private String ensureIndexed(Index proposedIndex) {
         }
 
         if (proposedIndex.indexType == IndexType.TEXT) {
-            if (!IndexManager.ftsAvailable(queue, database)) {
+            if (!((QueryableDatastore)datastore).ftsAvailable(queue)) {
```

Comment:
This cast can be removed if the constructor and local `datastore` field are updated.